### ISSUE TITLE
command 'java.edit.smartSemicolonDetectionUndo' not found when clicking backspace in Java file

### ISSUE THREAD

**Issue by testforstephen** [redhat-developer/vscode-java#3277]
This error happens in a certain time slot of loading projects.
![image](https://github.com/redhat-developer/vscode-java/assets/14052197/d77567d1-f83d-445f-a883-8f1ab3fd212f)

Here is the position we register the shortcut key `backspace` to the undo command when `javaLSReady` (javaLSReady indicates project initialized).

https://github.com/redhat-developer/vscode-java/blob/4352492f24acd0da086d6bc6e7dcd13567b82a70/package.json#L1333-L1337

Here is the position we register the undo command when ServiceReady.

https://github.com/redhat-developer/vscode-java/blob/4352492f24acd0da086d6bc6e7dcd13567b82a70/src/standardLanguageClient.ts#L124-L141

If you use backspace during the time slot between javaLSReady and serviceReady, you will encounter this issue.

**Comment by testforstephen** [comment]
As we agreed in our last meeting, can we just remove the command `java.edit.smartSemicolonDetectionUndo` and let the user undo the change directly with `ctrl + z`? Overriding the system shortcut `backspace` is a bit risky.

@snjeza @rgrunber @fbricon WDYT?

**Comment by fbricon** [comment]
yeah overriding backspace will certainly come bite us in the ass eventually.

Ctrl+Z is good enough

**Comment by rgrunber** [comment]
Agreed. However the type of problem you're seeing here might occur in other commands. I described it in https://github.com/redhat-developer/vscode-java/pull/3159#pullrequestreview-1541790270 . For larger projects there can be a gap between `Started` `ServiceReady`.

**Comment by rgrunber** [comment]
If any request to re-introduce "undo" come up we should probably point them to this discussion. We could definitely consider re-introducing it as its own option, but I'd first want to see smart semicolon detection itself enabled by default once we feel it's stable.
